cluster-autoscaler/core/static_autoscaler_test.go

```diff
@@ -991,6 +991,118 @@ func TestStaticAutoscalerRunOnceWithFilteringOnUpcomingNodesEnabledNoScaleUp(t *
 		podDisruptionBudgetListerMock, daemonSetListerMock, onScaleUpMock, onScaleDownMock)
 }
 
+func TestStaticAutoscalerRunOnceWithSchedulerProcessingIgnored(t *testing.T) {
```

Comment:
Can you make this test table-based? Possibly refactoring some of the existing tests.

Reply:
Updated tests & addsed a new function `testAutoscaler` to be used in other tests as well, also added a todo to refactor old tests to use the new `testAutoscaler` function

Reply:
I'm not sure if one huge testing function is the way to go. The actual assertions/checks belong in the test function, not some humongous utility. If you want to abstract some of it, I think the test setup could be abstracted, but I'd make it clear what the testing function is actually testing (i.e. keep the checks there).